addons/web/static/src/js/core/dom.js

```diff
@@ -343,19 +343,38 @@ var dom = {
             // part, the button is disabled without any visual effect.
             $button.addClass('o_debounce_disabled');
             Promise.resolve(dom.DEBOUNCE && concurrency.delay(dom.DEBOUNCE)).then(function () {
-                $button.addClass('disabled').prop('disabled', true);
                 $button.removeClass('o_debounce_disabled');
-
-                return Promise.resolve(result).then(function () {
-                    $button.removeClass('disabled').prop('disabled', false);
-                }).guardedCatch(function () {
-                    $button.removeClass('disabled').prop('disabled', false);
-                });
+                dom.addButtonLoadingEffect($button[0], true);
+                const always = () => {
+                    dom.addButtonLoadingEffect($button[0]);
+                };
```

Comment:
You could do that one line ;)
```suggestion
                const always = () => dom.addButtonLoadingEffect($button[0]);
```

But in fact I am not a fan of that `spin` parameter especially with it being `false` by default (calling a add function removing something). Here what I probably would have done is name the method "toggle" instead of "add" to follow JS conventions and make the spin parameter non optional. Or, I personally like something like that:

```javascript
const always = dom.addButtonLoadingEffect($button[0]);
```
=> No need of a parameter, the function returns a callback which is able to restore to the initial state.

But anything you want is fine here.

Reply:
As discussed I implemented it that way, it is **way** better.

Reply:
Not sure if ironic or not :stuck_out_tongue_winking_eye:

Reply:
No it is not. It is like 90% less verbose :+1:

Reply:
et 3x  moins usine a gaz, vu que 3x moins de loc ;)

Reply:
@JKE-be :grimacing: